## 9.01 Lab - Bayes Practice

> Author: Matt Brems

This lab will have two parts.

- Part 1: Bayes Practice Problems
- Part 2: Moving from Probabilities to Distributions

### Part 1: Bayes Practice Problems

##### Problem 1. You work at a financial services company that offers individual retirement accounts (IRAs). To target high potential customers, you want to find out if people with children are more likely to invest in IRAs.  

So far you have found:
- 30% of all Americans have IRAs.
- half of Americans have children.
- two-thirds of those with IRAs have children.

Based on this information, what is the probability that someone with children has an IRA? 

50% have children

66.6% of 30% have children

Only 30% have IRAs

A = Has IRA

P(A) = .3

B = Has children

P(B) = .5

P(B|A) = .6666...

In [1]:
(.3 * (2/3)) / .5

0.39999999999999997

**Answer:** 40%

---

##### Problem 2: Suppose I own a hotel and an apartment building. Times are tight, and I need to sell one of them. Both would earn me the exact same amount of money, so that doesn't help me to make my decision. Instead, I decide to make my decision based on my residents' satisfaction. 
- Whichever building has a higher percentage of satisfied people is the one I will choose to keep.
- _However_, if the percent of satisfied guests for my hotel and the percent of satified guests for my apartment building are within 10% of one another, I will sell neither until I can bring in some outside consultants to help me make my decision.
- Last winter, I administered a survey to *all* residents and found that 60% of them are satisfied.
- Among satisfied respondents, two out of every five came from the apartment building.
- One-third of my residents live in my apartment building.

What action will I take?

33.3% live in the apartments

66.6% live in the hotel (just roll with it)

60% are satisfied

2/5 of 60% live in the apartment

3/5 of 60% live in the hotel

A = Is satisfied

P(A) = .6

B = Lives in apts.

P(B) = .333

P(B|A) = .4

B` = Lives in hotel

P(B`) = .666

P(B`|A) = .6

In [2]:
(.4 * .6) / (1 / 3)

0.72

In [3]:
(.6 * .6) / (2 / 3)

0.54

**Answer:** Keep the apartment

---

##### Problem 3: Suppose you and your friend are playing a game. Your friend has laid four coins out in front of you. If you flip heads, you win a dollar from your friend. If you flip tails, you owe a dollar to your friend. However, the coins in front of you are not fair.
- One coin has a 80% chance of flipping heads. (Call this coin A.)
- One coin has a 60% chance of flipping heads. (Call this coin B.)
- One coin has a 40% chance of flipping heads. (Call this coin C.)
- One coin has a 10% chance of flipping heads. (Call this coin D.)

##### Problem 3-a) Suppose you select one coin at random. That is, you don't know whether you selected coin A, B, C, or D. You flip heads. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: When finding the marginal probability of flipping heads, you will need to use the Law of Total Probability! Find this **first**.

In [4]:
PA = ((.25 * .8) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1)))
PB = ((.25 * .6) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1)))
PC = ((.25 * .4) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1)))
PD = ((.25 * .1) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1)))
print(PA, PB, PC, PD)

0.4210526315789474 0.3157894736842105 0.2105263157894737 0.052631578947368425


**Answer**: A: 42.1% B: 31.6% C: 21.1% D: 5.3%

##### Problem 3-b) Suppose you are using the same coin as before. That is, you _still_ don't know whether you selected coin A, B, C, or D - although you have a better idea now that you flipped heads on your first flip! On this second flip, you flip tails. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: Everything now is **conditional on the fact that we have already flipped a heads**.

$$
\begin{eqnarray*}
P(A|B\cap C) &=& \frac{P(A\cap B \cap C)}{P(B\cap C)} \\
\\
&=& \frac{P(A\cap B|C)P(C)}{P(B|C)P(C)} \\
\\
&=& \frac{P(A\cap B|C)}{P(B|C)} \\
\end{eqnarray*}
$$

In [5]:
TandH = ((((.25 * .8) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1))) * .2)
         + (((.25 * .6) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1))) * .4)
         + (((.25 * .4) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1))) * .6)
         +  (((.25 * .1) / ((.25 * .8) + (.25 * .6) + (.25 * .4) + (.25 * .1))) * .9))
print(TandH)

0.3842105263157895


In [6]:
print((PA * .2) / TandH)
print((PB * .4) / TandH)
print((PC * .6) / TandH)
print((PD * .9) / TandH)

0.21917808219178084
0.3287671232876712
0.3287671232876712
0.1232876712328767


**Answer**: A: 21.9% B: 32.9% C: 32.9% D: 12.3%

##### BONUS: Problem 3-c) Suppose you are using the same coin as before. That is, you _still_ don't know whether you selected coin A, B, C, or D - although you have a better idea now that you flipped heads, then tails on your first two flips! On this third flip, you flip tails. Given this data, what are the probabilities that you selected coin A, coin B, coin C, and coin D?

> Hint: This will follow the same structure as the last two problems.

In [7]:
TandTandH = (TandH * .2) + (TandH * .4) + (TandH * .6) + (TandH * .9)
print(TandTandH)

0.8068421052631579


In [10]:
print((PA) / TandTandH)
print((PB) / TandTandH)
print((PC) / TandTandH)
print((PD) / TandTandH)

0.5218525766470972
0.3913894324853229
0.2609262883235486
0.06523157208088715


In [24]:
32/353

0.0906515580736544

**Answer**: A: 9.1%

---

### Part 2: Moving from Probabilities to Distributions

##### Problem 4. We want to look at our prior probabilities.

##### Problem 4-a) In Problem 3, before you had flipped any coin, what was the probability that you had selected coin A? Coin B? Coin C? Coin D?

**Answer:** 25%

##### Problem 4-b) What is the definition of a distribution?
> Check your notes from the early weeks of the course if you can't remember.

**Answer:** Set of observed outcomes and their frequency

##### Problem 4-c) What "named distribution" (i.e. a distribution that has a common name) could we apply to a situation where selecting coins A-D is equally likely?

**Answer:** Uniform discrete

##### Problem 4-d) Suppose instead of 4 coins, we have 10 or 20 or 100 coins. We can leverage the named distribution from 4-c) to each of these cases. If I have $k$ coins and use a Discrete Uniform distribution to model it, what is the probability of each coin being selected?

**Answer:** 1/k

##### Problem 4-e) When would it be impossible for us to list out all possible options? In these cases, specifying a distribution might be necessary!

**Answer:** If there are infinite possibilities

##### Problem 4-f) Suppose I have only one coin with some unknown probability of flipping heads $p$. If I want to specify some prior distribution where all probabilities $p$ are equally likely, then what named distribution might I use?

**Answer:** Continuous uniform

##### Problem 4-g) Suppose I have only one coin with some unknown probability of flipping heads $p$. If I think 50% is the likeliest value for $p$ and, as we get farther away from 50%, that value is less and less likely, what named distribution could I use?

**Answer:** Truncated Gaussian

##### Problem 4-g) When we enumerated our prior probabilities, we got the same number of posterior probabilities. For example, when we had four prior probabilities (one for Coin A, one for Coin B, one for Coin C, and one for Coin D), then we had four posterior probabilities (one for Coin A, for Coin B, one for Coin C, and one for Coin D). However, if we have one prior distribution, then we will get one posterior distribution. What are some advantages to being able to summarize our posterior with one distribution instead of a series of probabilities?

**Answer:** You can deal with continuous cases or with instances where the number of outcomes is vast